File: DesktopCore/src/bean/ScrollTableButton.java
package bean;

import javax.swing.Action;
import javax.swing.table.DefaultTableModel;

public class ScrollTableButton extends ScrollTable
{
	private int buttonColumn = -1;
	
	public ScrollTableButton(int buttonColumn, DefaultTableModel tm, Action a)
	{
		this.buttonColumn = buttonColumn;
		
		setModel(tm);
		ButtonColumn bc = new ButtonColumn(this, a, buttonColumn);
	}
	
	@Override
	public boolean isEditable(int row, int col)
	{
		return col == buttonColumn;
	}
}

File: src/main/kotlin/dartzee/core/bean/ScrollTableButton.kt
package dartzee.core.bean

import javax.swing.Action
import javax.swing.table.DefaultTableModel

class ScrollTableButton(tm: DefaultTableModel) : ScrollTable() {
    private val buttonColumns = mutableListOf<Int>()

    init {
        model = tm
    }

    override fun isEditable(row: Int, col: Int) = buttonColumns.contains(col)

    fun setButtonColumn(column: Int, action: Action) {
        buttonColumns.add(column)
        ButtonColumn(this, action, column)
    }
}
